scripts/remove_nearly_empty_products.pl

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2023 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ProductOpener::PerlStandards;
+
+use CGI::Carp qw(fatalsToBrowser);
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Products qw/retrieve_product/;
+use ProductOpener::Store qw/:all/;
+use ProductOpener::Data qw/get_products_collection/;
+
+# This script is run daily to remove empty products (without data or pictures)
+# in particular products created by the button to add a product without a barcode
+
+my $cursor = get_products_collection()->query({data_quality_errors_tags => { '$ne' => ''} })->fields({code => 1});
```

Comment:
This query is quite broad (any product with data quality errors), it will fetch ~ 150k products, that you will then read individually. You could change your query to put some of the filters in.
e.g. adding state_tags  en:photos-to-be-uploaded would divide the number of results by 10.

Reply:
I have tried `last_image_t => { '$eq' => '' }` but it failed to load (time out I guess).

I haven't so much trust in `state_tags => 'en:photos-to-be-uploaded'` because sometimes (rarely) computation of state_tags seems to fail. I better trust `last_image_t` which is raw data, not computed, isn't it?

Reply:
I added `state_tags => 'en:photos-to-be-uploaded'`.